openproblems/tasks/denoising/methods/alra.py

```diff
@@ -45,9 +46,60 @@ def alra(adata, test=False):
                 raise
 
     # transform back into original space
+    # functions are reversed!
     Y = scprep.utils.matrix_transform(Y, np.square)
     Y = scprep.utils.matrix_vector_elementwise_multiply(Y, libsize, axis=0)
     adata.obsm["denoised"] = Y
 
     adata.uns["method_code_version"] = "1.0.0"
     return adata
+
+
+@method(
+    method_name="ALRA (log norm)",
+    paper_name="Zero-preserving imputation of scRNA-seq data using "
+    "low-rank approximation",
+    paper_reference="linderman2018zero",
+    paper_year=2018,
+    code_url="https://github.com/KlugerLab/ALRA",
+    image="openproblems-r-extras",
+)
+def alra_log(adata, test=False):
+    import numpy as np
+    import rpy2.rinterface_lib.embedded
+    import scipy
+    import scprep
+
+    # libsize and log norm
+    # lib norm
+    adata.obsm["train_norm"] = adata.obsm["train"]  # .todense()
```

Comment:
Please remove this line and just make `adata.obsm["train"]` the input to the next line.